src/config/config-factory.ts

```diff
@@ -1,24 +1,105 @@
-import { PartnerConfig } from '.';
+import { ConfigSystem, PartnerConfig } from '.';
 import kenyaConfig from './chis-ke';
 import togoConfig from './chis-tg';
 import civConfig from './chis-civ';
 import maliConfig from './chis-ml';
+import defaultConfig from './default-config';
+import path from 'path';
+import fs from 'fs';
 
-export const CONFIG_MAP: { [key: string]: PartnerConfig } = {
-  'CHIS-KE': kenyaConfig,
-  'CHIS-TG': togoConfig,
-  'CHIS-CIV': civConfig,
-  'CHIS-ML': maliConfig
-};
-
-export default function getConfigByKey(key: string = 'CHIS-KE'): PartnerConfig {
-  const usingKey = key.toUpperCase();
-  console.log(`Using configuration: ${key}`);
-  const result = CONFIG_MAP[usingKey];
-  if (!result) {
-    const available = JSON.stringify(Object.keys(CONFIG_MAP));
-    throw Error(`Failed to start: Cannot find configuration "${usingKey}". Configurations available are ${available}`);
+export default class ConfigFactory {
+  private config: PartnerConfig | null = null;
+  private static configFactoryInstance: ConfigFactory;
+  private filePath: string;
+
+  private DEFAULT_CONFIG_MAP: { [key: string]: PartnerConfig } = {
+    'CHIS-KE': kenyaConfig,
+    'CHIS-TG': togoConfig,
+    'CHIS-CIV': civConfig,
+    'CHIS-ML': maliConfig,
+    DEFAULT: defaultConfig
+  };
+
+  constructor() {
+    this.filePath = path.join(this.getConfigUploadDirectory(), 'config.json');
+  }
+
+  static getConfigFactory(): ConfigFactory {
+    if (!ConfigFactory.configFactoryInstance) {
+      ConfigFactory.configFactoryInstance = new ConfigFactory();
+    }
+
+    return ConfigFactory.configFactoryInstance as ConfigFactory;
   }
 
-  return result;
+  loadConfig(key?: string): PartnerConfig {
+
+    if (this.config) {
+      return this.config;
+    }
+
+    if (fs.existsSync(this.filePath)) {
+      this.config = this.readConfig();
+      console.log(`Using uploaded configuration: ${this.filePath}`);
+      return this.config;
+    }
+
+    //set default config
+    if (!key) {
+      console.log('no configuration found. Setting to default');
+      const defaultConfig = this.DEFAULT_CONFIG_MAP.DEFAULT;
+      return this.config = defaultConfig;
+    }
+
+    const usingKey = key.toUpperCase();
+    console.log(`Using configuration: ${key}`);
+    const result = this.DEFAULT_CONFIG_MAP[usingKey];
```

Comment:
```suggestion
    this.config = this.DEFAULT_CONFIG_MAP[usingKey];
```

This would be cleaner without the intermittent variable `result` imo